### ISSUE TITLE
Allow further options in dind

### ISSUE THREAD

**Issue by enolfc** [jupyterhub/binderhub#914]
The binderhub helm chart (when dind enabled) creates pods with fixed arguments:

```
        args:
          - dockerd
          - --storage-driver={{ .Values.dind.storageDriver }}
          - -H unix://{{ .Values.dind.hostSocketDir }}/docker.sock
```

For some deployments it may be necessary to add/change some of the options. Specifically in my case I need to specify a --mtu option for setting a smaller than the default docker MTU. It would be good to be able to specify extra options there.

**Comment by betatim** [comment]
Sounds like a sensible and useful extension. Is there a common pattern for doing this in the helm chart world? Something like `extraEnv` which is a frequently used key for extending the environment variables.

Maybe `extraArgs`?

Do you want to create a PR to implement this?

**Comment by jhamman** [comment]
Might I suggest we finish/merge #838 before adding more dind features? 😉 

+1 on adding something like extraArgs though.

**Comment by enolfc** [comment]
I can create a PR for this with the `extraArgs`

**Comment by betatim** [comment]
`extraArgs` is just a guess so a short amount of research what other helm charts in for example https://github.com/helm/charts/ do would be good before spending too much time implementing something.

**Comment by enolfc** [comment]
PR #916 should fix this issue.

I have found two main ways of using extraArgs in the stable helm charts:
- as a list, so they basically get copied to the arguments of the pod, or
- as a dict, where you create arguments in the form of --key=value. 

I opted for the first one as I don't have quite clear if there will be any args that do not fit the --key=value schema. Also this option seems to have problems with values == false
